libraries/chain/controller.cpp

```diff
@@ -6359,6 +6357,11 @@ void controller_impl::on_activation<builtin_protocol_feature_t::sync_call>() {
       add_intrinsic_to_whitelist( ps.whitelisted_intrinsics, "get_call_data" );
       add_intrinsic_to_whitelist( ps.whitelisted_intrinsics, "set_call_return_value" );
    } );
+
+#if defined(EOSIO_EOS_VM_RUNTIME_ENABLED) || defined(EOSIO_EOS_VM_JIT_RUNTIME_ENABLED)
```

Comment:
Personally I think it would be cleaner to move all the `#if` to `wasm_allocator_pool` internals. Then none would be needed sprinkled through-out. `wasm_allocactor_pool` would always exist, the methods would just be no-ops with the `#if` only in `wasm_alloc_pool.cpp`.

Reply:
Thanks. Will do that.

Reply:
I didn't mention it on a prior PR, but it seems like any ifdef on `EOSIO_EOS_VM_RUNTIME_ENABLED` is unnecessary since currently we effectively consider EOS VM a fallback that is always available. Sure, technically it may not always be the case in the future, but if it ever comes to that there will probably be significant refactoring anyways.

Reply:
Is `EOSIO_EOS_VM_JIT_RUNTIME_ENABLED` still necessary?

Reply:
In this case no since it's `||`, but in general yes because we don't have EOS VM JIT on, for example, ARM